src/Service/Connector/NameComProvider.php

```diff
@@ -0,0 +1,130 @@
+    private const DEV_BASE_URL = 'https://api.dev.name.com';
+
+    /**
+     * Order a domain name with the Gandi API.
+     *
+     * @throws \Exception
+     * @throws TransportExceptionInterface
+     * @throws DecodingExceptionInterface
+     */
+    public function orderDomain(Domain $domain, bool $dryRun = false): void
+    {
+        $ldhName = $domain->getLdhName();
+        if (!$ldhName) {
+            throw new \InvalidArgumentException('Domain name cannot be null');
+        }
+
+        $this->client->request(
+            'POST',
+            '/v4/domains',
+            (new HttpOptions())
+                ->setHeader('Accept', 'application/json')
+                ->setAuthBasic($this->authData['username'], $this->authData['token'])
+                ->setBaseUri($dryRun ? self::DEV_BASE_URL : self::BASE_URL)
+                ->setJson([
+                    'domain' => [
+                        [
+                            'domainName' => $domain->getLdhName(),
+                            // 'contacts' => [],
+                            // 'privacyEnabled' => false,
+                            'locked' => false,
+                            'autorenewEnabled' => false,
+                        ],
+                        // 'purchasePrice' => 0,
+                        'purchaseType' => 'registration',
+                        'years' => 1,
+                        // 'tldRequirements' => []
+                    ],
+                ])
+                ->toArray()
+        )->toArray();
+    }
+
+    public function verifySpecificAuthData(array $authData): array
+    {
+        $username = $authData['username'];
+        $token = $authData['token'];
+
+        if (
+            !is_string($username) || empty($username)
+            || !is_string($token) || empty($token)
+        ) {
+            throw new BadRequestHttpException('Bad authData schema');
+        }
+
+        return [
+            'username' => $authData['username'],
+            'token' => $authData['token'],
+        ];
+    }
+
+    public function isSupported(Domain ...$domainList): bool
+    {
+        return true;
+    }
+
+    protected function getSupportedTldList(): array
+    {
+        return [];
+    }
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    protected function getCachedTldList(): CacheItemInterface
+    {
+        return $this->cacheItemPool->getItem('app.provider.namecom.supported-tld');
+    }
+
+    /**
+     * @throws TransportExceptionInterface
+     */
+    protected function assertAuthentication(): void
+    {
+        try {
+            $response = $this->client->request(
+                'GET',
+                '/v4/hello',
+                (new HttpOptions())
+                    ->setHeader('Accept', 'application/json')
+                    ->setAuthBasic($this->authData['username'], $this->authData['token'])
+                    ->setBaseUri($this->kernel->isDebug() ? self::DEV_BASE_URL : self::BASE_URL)
```

Comment:
The base URL selection is inconsistent across providers. This might be a refactor to plan for a future PR.